cxx-squid/src/main/java/org/sonar/cxx/preprocessor/SourceCodeProvider.java

```diff
@@ -108,16 +110,10 @@ public File getSourceCodeFile(String filename, String cwd, boolean quoted) {
     return result;
   }
 
-  public String getSourceCode(File file) {
-    String code = null;
-    if (file.isFile()) {
-      try {
-        code = FileUtils.readFileToString(file);
-      } catch (java.io.IOException e) {
-        LOG.error("Cannot read contents of the file '{}'", file);
-      }
-    }
-
-    return code;
+  public String getSourceCode(File file) 
```

Comment:
done, ive pushed the encoding from the cxxconfiguration into here so that the same encodign is used to read also the include paths.

this is somehting that needs addressing in order to support different encodings from different files. however i think core atm does not support this either